server/src/services/QwixxLogic.ts

```diff
@@ -120,25 +184,60 @@ export default class QwixxLogic {
       player.submissionCount === 1 &&
       !this._dice.validColouredNumbers[colourToMark]?.includes(num)
     ) {
-      throw new Error(
-        "Number selected doesn't equal to sum of white die and coloured die."
-      );
-    }
-
-    if (!player.markNumber(colourToMark, num)) {
-      throw new Error("Invalid move.");
+      return {
+        isValid: false,
+        errorMessage: new Error(
+          "Number selected doesn't equal to sum of white die and coloured die."
+        ),
+      };
     }
 
-    if (
-      (player === this.activePlayer && player.submissionCount === 2) ||
-      (player !== this.activePlayer && player.submissionCount === 1)
-    ) {
-      player.markSubmitted();
+    /*add check for number being lower or higher than last checked number */
+    if (colourToMark === "red" || colourToMark === "yellow") {
+      if (num <= highestMarkedNumber) {
+        return {
+          isValid: false,
+          errorMessage: new Error(
+            "Number must be above the last marked number"
+          ),
+        };
+      }
+    } else if (colourToMark === "green" || colourToMark === "blue") {
+      if (num >= lowestMarkedNumber) {
+        return {
+          isValid: false,
+          errorMessage: new Error(
+            "Number must be below the last marked number"
+          ),
+        };
+      }
```

Comment:
A style thing but I find nested if statements combined with "else if"  harder to read. 
Could it have been written like:
```TypeScript
if(colourToMark === "red" || colourToMark === "yellow") && (num <= highestMarkedNumber){...}

if(colourToMark === "green" || colourToMark === "blue") && (num >= lowestMarkedNumber){...}
```

Not really important to change it.

Reply:
Seems like it got marked incorrectly but it's about that bit of code related to checking highest and lowest number.